File: src/main/java/championship/model/Player.java
package championship.model;

import java.util.Objects;

class Player {
    private String nickName;

    Player(String nickName) {
        if (nickName == null) {
            throw new IllegalArgumentException("nickName cannot be null");
        }
        this.nickName = nickName;
    }

    public String getNickName() {
        return nickName;
    }

    @Override
    public boolean equals(Object other) {
        if (!(other instanceof Player)) {
            return false;
        } else {
            Player otherPlayer = (Player) other;
            boolean equals = nickName.compareTo(otherPlayer.nickName) == 0;
            return equals;
        }
    }

    @Override
    public int hashCode() {
        return Objects.hash(getNickName());
    }

    @Override
    public String toString() {
        String res = "Player nickName=" + nickName;
        return  res;
    }
}

File: csharp/championship/Player.cs
using System;

namespace championship
{
    public class Player
    {
        private readonly string _nickname;

        public Player(string nickname)
        {
            if (nickname == null)
            {
                throw new ArgumentNullException ("nickname");
            }
            _nickname = nickname;
        }

        public string NickName 
        {
            get 
            {
                return _nickname;
            }
        }

        public override bool Equals(Object obj) 
        {
            return obj is Player && this._nickname == ((Player)obj)._nickname;
        }

        public override int GetHashCode() 
        {
            return _nickname.GetHashCode();
        }
    }
}
